test/metabase/api/alert_test.clj

```diff
@@ -23,653 +24,12 @@
-  {:style/indent 1}
-  [[db-binding collection-binding alert-binding card-binding] & body]
-  ;; I'm always getting the order of these two mixed up, so let's try to check that here
-  (when alert-binding
-    (assert (not= alert-binding 'card)))
-  (when card-binding
-    (assert (not= card-binding 'alert)))
-  `(do-with-alert-in-collection!
-    (fn [~(or db-binding '_) ~(or collection-binding '_) ~(or alert-binding '_) ~(or card-binding '_)]
-      ~@body)))
-
-;; This stuff below is separate from `with-alert-in-collection` above!
-(defn- do-with-alerts-in-a-collection!
-  "Do `f` with the Cards associated with `alerts-or-ids` in a new temporary Collection. Grant perms to All Users to that
-  Collection using `f`.
-
-  (The name of this function is somewhat of a misnomer since the Alerts themselves aren't in Collections; it is their
-  Cards that are. Alerts do not go in Collections; their perms are derived from their Cards.)"
-  [grant-collection-perms-fn! alerts-or-ids f]
-  (mt/with-non-admin-groups-no-root-collection-perms
-    (mt/with-temp [:model/Collection collection]
-      (grant-collection-perms-fn! (perms-group/all-users) collection)
-      ;; Go ahead and put all the Cards for all of the Alerts in the temp Collection
-      (when (seq alerts-or-ids)
-        (doseq [alert (t2/select :model/Pulse :id [:in (set (map u/the-id alerts-or-ids))])
-                :let  [card (#'models.pulse/alert->card alert)]]
-          (t2/update! :model/Card (u/the-id card) {:collection_id (u/the-id collection)})))
-      (f))))
-
-(defmacro ^:private with-alerts-in-readable-collection! [alerts-or-ids & body]
-  `(do-with-alerts-in-a-collection! perms/grant-collection-read-permissions! ~alerts-or-ids (fn [] ~@body)))
-
-(defmacro ^:private with-alerts-in-writeable-collection! [alerts-or-ids & body]
-  `(do-with-alerts-in-a-collection! perms/grant-collection-readwrite-permissions! ~alerts-or-ids (fn [] ~@body)))
-
-;;; +----------------------------------------------------------------------------------------------------------------+
-;;; |                                       /api/alert/* AUTHENTICATION Tests                                        |
-;;; +----------------------------------------------------------------------------------------------------------------+
-
-;; We assume that all endpoints for a given context are enforced by the same middleware, so we don't run the same
-;; authentication test on every single individual endpoint
-
-(deftest auth-tests
-  (is (= (get request/response-unauthentic :body)
-         (client/client :get 401 "alert")))
-
-  (is (= (get request/response-unauthentic :body)
-         (client/client :put 401 "alert/13"))))
-
-;;; +----------------------------------------------------------------------------------------------------------------+
-;;; |                                                 GET /api/alert                                                 |
```

Comment:
We're keeping a shim for this API, shouldn't we keep the tests?

Reply:
i added a new set of tests that replicate this